### PR TITLE
opencv: Adding gtk variant

### PR DESCRIPTION
**Pull request by arjun-raj-kuppala**

### CODE REVIEW

**Comment by tldahlgren** [comment]
@bvanessen Did you want to comment on this PR?

File: var/spack/repos/builtin/packages/opencv/package.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ class Opencv(CMakePackage, CudaPackage):
     components = [
         '1394', 'ade', 'aravis', 'arith_dec', 'arith_enc', 'avfoundation', 'clp',
         'cuda', 'eigen', 'ffmpeg', 'freetype', 'gdal', 'gdcm', 'gphoto2', 'gstreamer',
-        'halide', 'hpx', 'imgcodec_hdr', 'imgcode_pfm', 'imgcodec_pxm',
+        'gtk', 'halide', 'hpx', 'imgcodec_hdr', 'imgcode_pfm', 'imgcodec_pxm',
```
Comment (bvanessen) [inline var/spack/repos/builtin/packages/opencv/package.py:61]: Shouldn't this be tied into adding another dependency when the variant is enabled?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/opencv/package.py:61]: Yes, this needs to add a `gtkplus` dependency too. Also, it seems like this component is entirely hidden on macOS, so we may need to add a conflict for macOS.

Comment (arjun-raj-kuppala) [reply var/spack/repos/builtin/packages/opencv/package.py:61]: @bvanessen @adamjstewart incorporated the requested changes. Please check.

**Review by alalazo — DISMISSED** [review]
Style tests are failing

**Comment by adamjstewart** [comment]
Still needs a dependency on gtkplus

**Comment by arjun-raj-kuppala** [comment]
> Style tests are failing

Taken care of this with new patch

**Comment by arjun-raj-kuppala** [comment]
> Still needs a dependency on gtkplus

Added the gtkplus dependency

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**